dbt/include/trino/macros/materializations/incremental.sql

```diff
@@ -59,6 +61,54 @@
     )
 {%- endmacro %}
 
+{% macro trino__get_merge_sql(target, source, unique_key, dest_columns, predicates=none) -%}
+    {%- set predicates = [] if predicates is none else [] + predicates -%}
+    {%- set dest_cols_csv = get_quoted_csv(dest_columns | map(attribute="name")) -%}
+    {%- set dest_cols_csv_source = dest_cols_csv.split(', ') -%}
+    {%- set update_columns = config.get('merge_update_columns', default = dest_columns | map(attribute="quoted") | list) -%}
+    {%- set sql_header = config.get('sql_header', none) -%}
+
+    {% if unique_key %}
+        {% if unique_key is sequence and unique_key is not mapping and unique_key is not string %}
+            {% for key in unique_key %}
+                {% set this_key_match %}
+                    DBT_INTERNAL_SOURCE.{{ key }} = DBT_INTERNAL_DEST.{{ key }}
+                {% endset %}
+                {% do predicates.append(this_key_match) %}
+            {% endfor %}
+        {% else %}
+            {% set unique_key_match %}
+                DBT_INTERNAL_SOURCE.{{ unique_key }} = DBT_INTERNAL_DEST.{{ unique_key }}
+            {% endset %}
+            {% do predicates.append(unique_key_match) %}
+        {% endif %}
+    {% else %}
+        {% do predicates.append('FALSE') %}
+    {% endif %}
+
+    {{ sql_header if sql_header is not none }}
+
+    merge into {{ target }} as DBT_INTERNAL_DEST
```

Comment:
```
merge into "iceberg"."test16611829018484321781_test_incremental_merge"."trinary_unique_key_list" as DBT_INTERNAL_DEST
        using "iceberg"."test16611829018484321781_test_incremental_merge"."trinary_unique_key_list__dbt_tmp" as DBT_INTERNAL_SOURCE
        on 
                    DBT_INTERNAL_SOURCE.state = DBT_INTERNAL_DEST.state
                 and 
                    DBT_INTERNAL_SOURCE.county = DBT_INTERNAL_DEST.county
                 and 
                    DBT_INTERNAL_SOURCE.city = DBT_INTERNAL_DEST.city
                

    
    when matched then update set
        "state" = DBT_INTERNAL_SOURCE."state","county" = DBT_INTERNAL_SOURCE."county","city" = DBT_INTERNAL_SOURCE."city","last_visit_date" = DBT_INTERNAL_SOURCE."last_visit_date"
    

    when not matched then insert
        ("state", "county", "city", "last_visit_date")
    values
        (DBT_INTERNAL_SOURCE."state", DBT_INTERNAL_SOURCE."county", DBT_INTERNAL_SOURCE."city", DBT_INTERNAL_SOURCE."last_visit_date")
```

Cosmetically this output - for `test__trinary_unique_key_list` test - (even though purely generated) can be formatted to make it a bit easier to read by humans.

Reply:
Formatting changed to:
```
merge into "iceberg"."test16612448099870284150_test_incremental_merge"."trinary_unique_key_list" as DBT_INTERNAL_DEST 
using "iceberg"."test16612448099870284150_test_incremental_merge"."trinary_unique_key_list__dbt_tmp" as DBT_INTERNAL_SOURCE on
        DBT_INTERNAL_SOURCE.state = DBT_INTERNAL_DEST.state and
        DBT_INTERNAL_SOURCE.county = DBT_INTERNAL_DEST.county and
        DBT_INTERNAL_SOURCE.city = DBT_INTERNAL_DEST.city

        when matched then update set
            "state" = DBT_INTERNAL_SOURCE."state",
            "county" = DBT_INTERNAL_SOURCE."county",
            "city" = DBT_INTERNAL_SOURCE."city",
            "last_visit_date" = DBT_INTERNAL_SOURCE."last_visit_date"

        when not matched then insert
            ("state", "county", "city", "last_visit_date")
        values
            (DBT_INTERNAL_SOURCE."state", DBT_INTERNAL_SOURCE."county", DBT_INTERNAL_SOURCE."city", DBT_INTERNAL_SOURCE."last_visit_date")